runtime/autoload/ccomplete.lua

```diff
@@ -0,0 +1,892 @@
+    end
+
+    for _, tagidx in NVIM9.iter(NVIM9.fn['range'](NVIM9.fn['len'](diclist))) do
+      local nvim9_status, nvim9_ret = body(_, tagidx)
+      if nvim9_status == NVIM9.ITER_BREAK then
+        break
+      elseif nvim9_status == NVIM9.ITER_RETURN then
+        return nvim9_ret
+      end
+    end
+
+    if NVIM9.fn['len'](res) > 0 then
+      return NVIM9.ITER_BREAK
+    end
+
+    return NVIM9.ITER_DEFAULT
+  end
+
+  for _, tidx in NVIM9.iter(NVIM9.fn['range'](NVIM9.fn['len'](tokens))) do
+    local nvim9_status, nvim9_ret = body(_, tidx)
+    if nvim9_status == NVIM9.ITER_BREAK then
+      break
+    elseif nvim9_status == NVIM9.ITER_RETURN then
+      return nvim9_ret
+    end
+  end
+
+  return res
+end
+
+StructMembers = function(atypename, items, all)
+  all = NVIM9.bool(all)
+
+  -- # Search for members of structure "typename" in tags files.
+  -- # Return a list with resulting matches.
+  -- # Each match is a dictionary with "match" and "tagline" entries.
+  -- # When "all" is true find all, otherwise just return 1 if there is any member.
+
+  -- # Todo: What about local structures?
+  local fnames = NVIM9.fn['join'](NVIM9.fn['map'](
+    NVIM9.fn['tagfiles'](),
+    function(_, v)
+      return NVIM9.fn['escape'](v, ' \\#%')
+    end
+  ))
+  if fnames == '' then
+    return {}
+  end
+
+  local typename = atypename
+  local qflist = {}
+  local cached = 0
+  local n = ''
+  if NVIM9.bool(NVIM9.prefix['Bang'](all)) then
+    n = '1'
+    if NVIM9.bool(NVIM9.fn['has_key'](grepCache, typename)) then
+      qflist = NVIM9.index(grepCache, typename)
+      cached = 1
+    end
+  else
+    n = ''
+  end
+  if NVIM9.bool(NVIM9.prefix['Bang'](cached)) then
+    while 1 do
+      vim.api.nvim_command(
+        'silent! keepjumps noautocmd '
+          .. n
+          .. 'vimgrep '
+          .. '/\\t'
+          .. typename
+          .. '\\(\\t\\|$\\)/j '
+          .. fnames
+      )
+
+      qflist = NVIM9.fn['getqflist']()
+      if NVIM9.fn['len'](qflist) > 0 or NVIM9.fn['match'](typename, '::') < 0 then
+        break
+      end
+      -- # No match for "struct:context::name", remove "context::" and try again.
+      typename = NVIM9.fn['substitute'](typename, ':[^:]*::', ':', '')
+    end
+
+    if NVIM9.bool(NVIM9.prefix['Bang'](all)) then
+      -- # Store the result to be able to use it again later.
+      grepCache[NVIM9.index_expr(typename)] = qflist
+    end
+  end
+
+  -- # Skip over [...] items
+  local idx = 0
+  local target = ''
+  while 1 do
+    if idx >= NVIM9.fn['len'](items) then
+      target = ''
+      break
+    end
+    if NVIM9.index(NVIM9.index(items, idx), 0) ~= '[' then
+      target = NVIM9.index(items, idx)
+      break
+    end
+    idx = idx + 1
+  end
+  -- # Put matching members in matches[].
+  local matches = {}
+
+  for _, l in NVIM9.iter(qflist) do
+    local memb = NVIM9.fn['matchstr'](NVIM9.index(l, 'text'), '[^\\t]*')
+    if NVIM9.ops['RegexpMatches'](memb, '^' .. target) then
```

Comment:
would it be more intuitive to name this `vim9` instead of `NVIM9` ?

Reply:
i'd be fine with `vim9script` or `vim9`. I can prevent naming clashes in the generation step. I'll do whichever you prefer for merging.